pkg/manifestreader/path.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifestreader
+
+import (
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+// PathManifestReader reads manifests from the provided path
+// and returns them as Info objects. The returned Infos will not have
+// client or mapping set.
+type PathManifestReader struct {
+	Path string
+
+	ReaderOptions
+}
+
+// Read reads the manifests and returns them as Info objects.
+func (p *PathManifestReader) Read() ([]*resource.Info, error) {
```

Comment:
We might want to add that the mapping is not set in the Info (since the builder is `Local()`).